src/Core/src/Internal/Config/StateBinder.php

```diff
@@ -58,6 +59,13 @@ public function bind(string $alias, mixed $resolver): void
      */
     public function bindSingleton(string $alias, mixed $resolver): void
     {
+        if ($this->hasInstance($alias)) {
+            throw new DuplicateBindingException(\sprintf(
+                'Singleton `%s` already exists. To override a binding, call the `removeBinding` method first.',
```

Comment:
The message should be in the exception class and shouldnt contain a solution.
"Can't overload the singleton `%s` because it's already used."

Reply:
Thanks, changed.